gto/tag.py

```diff
@@ -183,16 +183,26 @@ def find(
     return tags
 
 
-def create_tag(repo, name, ref, message):
+def create_tag(
+    repo: git.Repo,
+    name: str,
+    ref: str,
+    message: str,
+    GIT_COMMITTER_NAME: str = None,
+    GIT_COMMITTER_EMAIL: str = None,
+):
     if all(c.hexsha != ref for c in repo.iter_commits()):
```

Comment:
Not about this PR, but looks dangerous. You are listing all the commits for existence test. Two more efficient approaches:
- `repo.commit(ref)` and catch a possible error
- try `repo.git.tag(...)` and catch an exception (probably less nice than the one from above)

Second should be more efficient, providing a zero overhead for a happy path.

Reply:
Thanks! Will do.